server/pbench/bin/pbench-audit-archive

```diff
@@ -0,0 +1,200 @@
+test -d $LOGSDIR || doexit "Bad logs directory: $LOGSDIR"
+
+# work files
+workdir=$TMP/$PROG.work.$$
+report=$workdir/report
+index_content=$workdir/index_content
+badcontrollers=$workdir/badcontrollers
+controllers=$workdir/controllers
+nonprefixes=$workdir/nonprefixes
+wrongprefixes=$workdir/wrongprefixes
+unexpectedsymlinks=$workdir/unexpectedsymlinks
+unexpectedfiles=$workdir/unexpectedfiles
+directories=$workdir/directories
+tarballs=$workdir/tarballs
+linkdirs=$workdir/linkdirs
+
+# make sure the directory exists
+mkdir -p $workdir
+
+trap "rm -rf $workdir" EXIT INT QUIT
+
+for ldir in $LINKDIRS; do printf "\t  ${ldir}\n"; done | sort > ${linkdirs}
+
+function verify_subdirs {
+    controller=${1}
+ 
+    let cnt=0
+
+    find ${controller} -maxdepth 1 -type d ! -name . ! -name $(basename ${controller}) ! -name .prefix -printf "\t  %f\n" | sort > ${directories} 2>&1
+    if [ -s ${directories} ]; then
+        grep -v WONT-INDEX ${directories} > ${directories}.linkdirs
+        comm -13 ${linkdirs} ${directories}.linkdirs > ${directories}.unexpected
+        if [ -s ${directories}.unexpected ]; then
+            printf "\t* Unexpected state directories found in this controller directory:\n"
+            printf "\t  ++++++++++\n"
+            cat ${directories}.unexpected
+            printf "\t  ----------\n"
+            let cnt=cnt+1
+        fi
+    else
+        printf "\t* No state directories found in this controller directory.\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${directories} ${directories}.linkdirs ${directories}.unexpected
+}
+
+function verify_tarball_names {
+    controller=${1}
+
+    let cnt=0
+
+    find ${controller} -maxdepth 1 -type l -printf "\t  %f -> %l\n" | sort > ${unexpectedsymlinks} 2>&1
+    if [ -s ${unexpectedsymlinks} ]; then
+        printf "\t* Unexpected symlinks in controller directory:\n"
+        printf "\t  ++++++++++\n"
+        cat ${unexpectedsymlinks}
+        printf "\t  ----------\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${unexpectedsymlinks}
+
+    find ${controller} -maxdepth 1 -type f ! -name '*.tar.xz.md5' ! -name '*.tar.xz' -printf "\t  %f\n" | sort > ${unexpectedfiles} 2>&1
+    if [ -s ${unexpectedfiles} ]; then
+        printf "\t* Unexpected files in controller directory:\n"
+        printf "\t  ++++++++++\n"
+        cat ${unexpectedfiles}
+        printf "\t  ----------\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${unexpectedfiles}
+
+    find ${controller} -maxdepth 1 -type f \( -name '*.tar.xz.md5' -o -name '*.tar.xz' \) > ${tarballs} 2>&1
+    if [ ! -s ${tarballs} ]; then
+        printf "\t* No tar ball files found in this controller directory.\n"
+        let cnt=cnt+1
+    fi
+    rm -f ${tarballs}
+
+    return $cnt
+}
+
+function verify_prefixes {
+    controller=${1}
+
+    if [ ! -e ${controller}/.prefix ]; then
+        return 0
+    fi
+    if [ ! -d ${controller}/.prefix ]; then
+        printf "\t* Prefix directory, .prefix, is not a directory\n"
```

Comment:
"... ${controller}/.prefix ..." ?

Reply:
Line 180 emits the controller name before these reports, so was hoping to not duplicate information if possible.